cpp/src/arrow/flight/client.cc

```diff
@@ -531,14 +531,19 @@ class FlightClient::FlightClientImpl {
 
     std::stringstream grpc_uri;
     std::shared_ptr<grpc::ChannelCredentials> creds;
-    if (scheme == kSchemeGrpc || scheme == kSchemeGrpcTcp || scheme == kSchemeGrpcTls) {
+    if (scheme == kSchemeGrpc || scheme == kSchemeGrpcTcp || scheme == kSchemeGrpcTls || scheme == kSchemeGrpcMTls) {
       grpc_uri << location.uri_->host() << ":" << location.uri_->port_text();
 
-      if (scheme == "grpc+tls") {
+      if ((scheme == "grpc+tls") || (scheme == "grpc+mtls") {
         grpc::SslCredentialsOptions ssl_options;
         if (!options.tls_root_certs.empty()) {
           ssl_options.pem_root_certs = options.tls_root_certs;
         }
+        if (scheme  == 'grpc+mtls'){
```

Comment:
Is a whole new scheme necessary? I'd say that so long as the client has provided a list of client cert/key pairs, then mTLS should just be enabled, no?

Reply:
Also, this doesn't build and needs to be formatted - the developer guide has instructions: https://arrow.apache.org/docs/developers/cpp/index.html